internal/tools/aws/database_multitenant.go

```diff
@@ -92,6 +93,45 @@ func (d *RDSMultitenantDatabase) MaxSupportedDatabases() int {
 	return DefaultRDSMultitenantDatabasePostgresCountLimit
 }
 
+// RefreshResourceMetadata ensures various multitenant database resource's
+// metadata are correct.
+func (d *RDSMultitenantDatabase) RefreshResourceMetadata(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	return d.updateMultitenantDatabase(store, logger)
+}
+
+func (d *RDSMultitenantDatabase) updateMultitenantDatabase(store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) error {
+	database, unlockFn, err := d.getAndLockAssignedMultitenantDatabase(store, logger)
+	if err != nil {
+		return errors.Wrap(err, "failed to get and lock assigned database")
+	}
+	if database == nil {
+		return errors.Wrap(err, "failed to find assigned multitenant database")
+	}
+	defer unlockFn()
+
+	logger = logger.WithField("assigned-database", database.ID)
+
+	rdsCluster, err := d.describeRDSCluster(database.ID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to describe the multitenant RDS cluster ID %s", database.ID)
+	}
+
+	weight, err := store.GetInstallationsTotalDatabaseWeight(database.Installations)
+	if err != nil {
+		return errors.Wrap(err, "failed to calculate total database weight")
+	}
+	roundedUpWeight := int(math.Ceil(weight))
+
+	err = d.updateCounterTag(rdsCluster.DBClusterArn, roundedUpWeight)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update tag:counter in RDS cluster ID %s", *rdsCluster.DBClusterIdentifier)
+	}
+
+	logger.Debugf("Multitenant database %s counter value updated to %d", database.ID, roundedUpWeight)
```

Comment:
I believe we use the exact same logic in 3 places now. I would consider making it a separate function.

Reply:
Thanks for the suggestion. I originally didn't make a helper function for this due to the differences in locking, but I just realized I can at least split out the tag part. Thanks!